### PR TITLE
[ROCm][CI] Validate Mooncake and NIXL prefill/decode accuracy

### PR DESCRIPTION
**Pull request by AndreasKaratzas**
- Add two Mooncake HIP and six NIXL/AITER cases, preserving upstream model/TP pairs, all 1,319 GSM8K questions, 100 concurrent requests and the original accuracy bounds.
- Pin model/tokenizer revisions and require positive evaluation-only external-KV deltas; also require transferred bytes for NIXL, with explicit cache bounds and owned-process cleanup.
- Add MI355 native mirrors and matching optional legacy jobs, preserving the existing NVIDIA and generic Triton NIXL selections.

The existing AMD generic NIXL sweep already covers these model and tensor-parallel combinations with Triton attention. The missing upstream counterparts exercise a second attention backend and Mooncake transfer, so these tests use AITER unified attention and native Mooncake HIP transport. Reusing the original accuracy functions preserves their full workload and acceptance bounds. Evaluation-only counters establish that transferred KV contributes during GSM8K beyond the preliminary smoke request.

AI assistance: OpenAI Codex assisted with implementation and validation.

### CODE REVIEW

**Comment by chilkotiKartik** [comment]
Review for PR #58095: [ROCm][CI] Validate Mooncake and NIXL prefill/decode accuracy

### Distributed CI & Accuracy Validation

1. **Numerical Invariants**:
   Evaluating logit cosine similarity and cross-entropy divergence across disaggregated Prefill/Decode nodes using Mooncake and NIXL transport layers provides a robust safeguard against RDMA memory transposition errors.

2. **ROCm Memory Synchronization**:
   Ensure hipMemcpyAsync and RCCL collective operations synchronize before logit assertions run in the CI test harness to avoid race conditions on high-throughput MI300X clusters.

File: tests/v1/kv_connector/rocm_pd_accuracy_utils.py [hunk 1]
```diff
@@ -0,0 +1,274 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (tjtanaa) [inline tests/v1/kv_connector/rocm_pd_accuracy_utils.py:1]: I would recommend if possible, we split this up into 

`tests/v1/kv_connector/mooncake_integration` and `tests/v1/kv_connector/nixl_integration` like on CUDA to keep the tests in same organization structure.

Image

Comment (AndreasKaratzas) [reply tests/v1/kv_connector/rocm_pd_accuracy_utils.py:1]: @tjtanaa Done :)

**Review by tjtanaa — APPROVED** [review]
Lgtm
